opal-core-ws/src/main/java/org/obiba/opal/web/analysis/TableAnalysisResource.java

```diff
@@ -0,0 +1,124 @@
+                                AnalysisExportService analysisExportService) {
+
+    this.analysisService = analysisService;
+    this.analysisResultService = analysisResultService;
+    this.analysisExportService = analysisExportService;
+  }
+
+  @GET
+  @Path("analyses")
+  public Projects.OpalAnalysesDto getProjectTableAnalyses(@QueryParam("order") @DefaultValue("desc") String order,
+                                                          @QueryParam("limit") @DefaultValue("0") int limit) {
+    return Projects.OpalAnalysesDto.newBuilder()
+      .addAllAnalyses(
+        StreamSupport
+          .stream(analysisService.getAnalysesByDatasourceAndTable(datasourceName, tableName, order, limit).spliterator(), false)
+          .map(analysis -> Dtos.asDto(analysis).build()).collect(Collectors.toList()))
+        .build();
+  }
+
+  @GET
+  @Path("/analysis/{analysisId}")
+  public OpalAnalysisDto getAnalysis(@PathParam("analysisId") String analysisId,
+                                     @QueryParam("order") @DefaultValue("desc") String order,
+                                     @QueryParam("limit") @DefaultValue("0") int limit) {
+    Builder builder = Dtos.asDto(analysisService.getAnalysis(analysisId));
+
+    builder.addAllAnalysisResults(
+      StreamSupport
+        .stream(analysisResultService.getAnalysisResults(analysisId, order, limit).spliterator(), false)
+        .map(analysisResult -> Dtos.asDto(analysisResult).build())
+        .collect(Collectors.toList()));
+
+    return builder.build();
+  }
+
+  @GET
+  @Path("/analysis/{analysisId}/results")
+  public OpalAnalysisResultsDto getAnalysisResults(@PathParam("analysisId") String analysisId,
+                                                   @QueryParam("order") @DefaultValue("desc") String order,
+                                                   @QueryParam("limit") @DefaultValue("0") int limit) {
+    return OpalAnalysisResultsDto.newBuilder()
+        .addAllAnalysisResults(
+          StreamSupport
+            .stream(analysisResultService.getAnalysisResults(analysisId, order, limit).spliterator(), false)
+            .map(analysisResult -> Dtos.asDto(analysisResult).build()).collect(Collectors.toList()))
+        .build();
+  }
+
+  @GET
+  @Path("/analysis/{analysisId}/result/{rid}")
+  public OpalAnalysisResultDto getAnalysisResult(@PathParam("analysisId") String analysisId, @PathParam("rid") String rid) {
+    return Dtos.asDto(analysisResultService.getAnalysisResult(analysisId, rid)).build();
+  }
+
+
+  @GET
+  @Path("analyses/_export")
+  @Produces("application/zip")
+  public Response exportTableAnalysis(@QueryParam("aOrder") @DefaultValue("desc") String aOrder,
+                                      @QueryParam("aLimit") @DefaultValue("0") int aLimit,
+                                      @QueryParam("rOrder") @DefaultValue("desc") String rOrder,
+                                      @QueryParam("rLimit") @DefaultValue("0") int rLimit) {
```

Comment:
When ``limit == 0`` the list is empty?

Reply:
If you download a bunch of files, then you do not care about the order. This web service(s) should be simplified by only offering the ability to get the latest (default) or all (option that will be never used in my opinion).